src/services/OswServiceTaskConsole.cpp

```diff
@@ -0,0 +1,178 @@
+#include "./services/OswServiceTaskConsole.h"
+#include "osw_hal.h"
+
+void OswServiceTaskConsole::setup() {
+    OswServiceTask::setup();
+    OSW_LOG_I("Console is now enabled.");
+#ifdef NDEBUG
+    this-> m_locked = true; // in release mode the console is locked by default
+#endif
+    this->newPrompt();
+}
+
+void OswServiceTaskConsole::loop() {
+    while(true) {
+        if(!Serial.available()) break;
+        char c = Serial.read();
+        if(32 <= c and c <= 126) { // printable characters
+            this->m_inputBuffer += c;
+            Serial.print(c); // echo the entered character back
+        } else if(c == 10 or c == 13) { // CR or LF
+            this->runPrompt();
+            this->newPrompt();
+        } else if(c == 8 or c == 127 or c == 27) { // backspace / delete / escape
+            if(this->m_inputBuffer.length() > 0) {
+                this->m_inputBuffer.pop_back();
+            }
+            Serial.print(c);
+        } else if(c == 9) { // tab
+            Serial.println(); // finish the prompt line
+            this->showHelp();
+            this->showPrompt();
+        } else {
+            Serial.print((char) 0x07); // bell
+            OSW_LOG_D("Unprocessable character (",(int) c, "): ", c);
+        }
```

Comment:
how about switch-case?
and IMHO, unless it's 1 and 0, all hard-coded constants would be read is difficult better.. if they had an alternative preprocessing variable. Even if you add an annotation comment `// blah blah`..
```
TAB = 9
{ . . . }
c == TAB
```

Reply:
Hmm, the first case cannot be done via a switch case, as it covers a range. The remaining yes, while I still feel this will bloat up the code, it can be done (especially thinking of potential optimizations). I'll include it. :+1:

Reply:
Done